addons/website_crm/controllers/main.py

```diff
@@ -80,6 +80,12 @@ def insert_record(self, request, model, values, custom, meta=None):
                         values['partner_id'] = visitor_partner.id
                 else:
                     values['partner_id'] = visitor_partner.id
+            elif values_email_normalized and not visitor_partner:
+                email_partner = request.env['res.partner'].sudo().search([('email_normalized', '=', values_email_normalized)], limit=1)
+                email_user = request.env['res.users'].sudo().search([('partner_id', 'in', email_partner.ids)], limit=1)
```

Comment:
What if there is no partner? It will work since partner_id is required and it will return nothing (in []) but wouldn't it better in a `if email_partner:` ?